example/ipss_app/parse_and_plot.py

```diff
@@ -0,0 +1,208 @@
+    """
+    Perform a simple binning on the collected spikes before duration
+    Assuming the times are in ms resolution
+    bin_size (ms)
+    duration (ms)
+    """
+    n_bins = int(math.floor(duration / bin_size))
+    bins=[0]*n_bins
+
+    for gid, spikes in spikes_per_cell.items():
+        for spike in spikes:
+            # Skip late spikes
+            if spike > duration:
+                continue
+
+            # Convert to bin idx and increase 
+            bins[int(math.floor(spike / bin_size))] += 1
+
+    # normalize to spikes/second. 
+    # we have 10000 cells but are binning per 1000
+    # so devide by 10
+    bins = [x / 10 for x in bins]
+
+    return bins  
+
+def plot_histogram_and_target_curve(bins, times=None, rates=None, plot_target=True):
+    """
+    Plot the histogram and target curve in one figure
+    """
+    plt.hist([idx for idx in range(len(bins))], len(bins), weights = bins)
+    if plot_target:
+        plt.plot(times, rates, linewidth=7,alpha=0.7)
+
+    plt.title("Instantanious spike rate for a population of ipss cells ")
+    plt.xlabel("Time (ms.)")
+    plt.ylabel("Instantanious spike rate (spikes / second)")
+    plt.show()
+
+def main(path_spikes=None, path_time_rate=None, plot_target=True, interpolate=True):
+    """
+    Simple main function
+    - Parse the spikes (expected in gdf format) from path 
+      if path == None default path './spikes.gdf' will be used
+    - Bin the spikes in 1000 ms bins
+    - Plot a histogram
+    - Plot a line with the default / expected rate if path is supplied
+    todo:
+    - Use the power of statistics to make this an automatic test
+    """
+    plot_default_curve = True
+
+    # If no path supplied 
+    if path_spikes == None:
+        path_spikes = default_spike_file_path
+
+    # If no path supplied 
+    if not path_time_rate == None:
+        time_rates = parse_time_rate(path_time_rate)
+    else:
+        time_rates = default_time_rate_pairs
+
+
+    # parse the spikes
+    spikes_per_cell = parse_spike_file(path_spikes)
+
+    # perform binnen
+    bins = binning(spikes_per_cell)
+    
+    # convert time_rate pairs to two lists
+    # if we are NOT interpolating we need to fix the time_rates pairs
+    if not interpolate:
+        time_rates = [x for x in time_rates for _ in (0, 1)]  # double each entry
+        times, rates = zip(*time_rates)
+
+        # add on entry in front and back for non interpolated curve
+        rates = [rates[0]] + list(rates)
+        times = list(times) + [times[-1]]
+    else:
+        times, rates = zip(*time_rates)
+
+    # plot the binned spikes and the optional curve
+    plot_histogram_and_target_curve(bins, times, rates, plot_target)
+
+def evaluate_string_true(s="True"):
+    """
+    Evaluates if s is a string with True value 
+    True -> 'true', '1', 't', 'yes', 'y'
+    Capitalization is ignored
+    """
+    return s.lower() in ['true', '1', 't', 'yes', 'y']
```

Comment:
Don't know whether a tuple would be implemented better than a list here, since the tuple is fixed length.